lxqt-session/src/lxqtmodman.cpp

```diff
@@ -319,7 +319,25 @@ void LXQtModuleManager::restartModules(int exitCode, QProcess::ExitStatus exitSt
 LXQtModuleManager::~LXQtModuleManager()
 {
     qApp->removeNativeEventFilter(this);
-    qDeleteAll(mNameMap);
+
+    // We disconnect the finished signal before deleting the process. We do
+    // this to prevent a crash that results from a state change signal being
+    // emmited while deleting a crashing module.
+    // If the module is still connect restartModules will be called with a
+    // invalid sender.
+
+    ModulesMapIterator i(mNameMap);
+    while (i.hasNext())
+    {
+        i.next();
+
+        auto p = i.value();
+        disconnect(p, SIGNAL(finished(int, QProcess::ExitStatus)), 0, 0);
+
+        delete p;
+        p = 0;
```

Comment:
Thanks @palinek. That's consequence of using Java style iterators. I also don't like them. Actually I do not use them. Just following the class style.
We can change it. But IMO all the other Java style iterators should also be changed.